src/test/java/com/timgroup/statsd/UnixSocketTest.java

```diff
@@ -62,23 +67,37 @@ public void start() throws IOException {
         socketFile = new File(tmpFolder, "socket.sock");
         socketFile.deleteOnExit();
 
-        server = new UnixSocketDummyStatsDServer(socketFile.toString());
+        if (transport.equals("unixgram")) {
+            server = new UnixDatagramSocketDummyStatsDServer(socketFile.toString());
+        } else {
+            server = new UnixStreamSocketDummyStatsDServer(socketFile.toString());
+        }
+
+        Callable<SocketAddress> addressLookup = new Callable<SocketAddress>() {
+            @Override
+            public SocketAddress call() throws Exception {
+                return new UnixSocketAddressWithTransport(new UnixSocketAddress(socketFile.getPath()), UnixSocketAddressWithTransport.TransportType.fromScheme(transport));
+            }
+        };
+
         client = new NonBlockingStatsDClientBuilder().prefix("my.prefix")
-            .hostname(socketFile.toString())
+            .addressLookup(addressLookup)
```

Comment:
Probably, but I was thinking that we should do that when we want to actually advertise this more widely. Currently the feature is still considered experimental and not documented. What do you think ?

Reply:
I think it may be better to document it and make it explicit that the feature is experimental and may be subject to breaking changes, otherwise it just looks like an oversight. I'll leave it to you whether to add a builder method.

Reply:
Added a method to the builder